api/v1beta1/hetznercluster_webhook.go

```diff
@@ -109,6 +149,55 @@ func (r *HetznerCluster) ValidateCreate() (admission.Warnings, error) {
 		if err := isNetworkZoneSameForAllRegions(r.Spec.ControlPlaneRegions, nil); err != nil {
 			allErrs = append(allErrs, err)
 		}
+	} else {
+		// If ID is given check that all other network settings are empty.
+		if r.Spec.HCloudNetwork.ID != nil {
+			if errs := areCIDRsAndNetworkZoneEmpty(r.Spec.HCloudNetwork); errs != nil {
+				allErrs = append(allErrs, errs...)
+			}
+		} else {
+			// If no ID is given check the other network settings for valid entries.
+			if r.Spec.HCloudNetwork.NetworkZone != nil {
```

Comment:
this is a validation that we don't have right now I think. It seems to me as if we should add a separate PR for that, as this part is relevant independent of the purpose of this PR!

Reply:
I think it's good to have it.

Reply:
I assume there was validation for this before, because of following kubebuilder markers, which would enforce the API server to perform the validations for the resource:
```
	// +kubebuilder:validation:Enum=eu-central;us-east;us-west;ap-southeast
	// +kubebuilder:default=eu-central
```

I think that the manual validation is now necessary because of the validation now being dependent on the new ID field.